BGAnimations/ScreenSystemLayer overlay.lua

```diff
@@ -210,4 +210,61 @@ t[#t+1] = LoadFont("Common Footer")..{
 	end
 }
 
+-- -----------------------------------------------------------------------
+-- Modules
+
+local function LoadModules()
+	-- A table that contains a [ScreenName] -> Table of Actors mapping.
+	-- Each entry will then be converted to an ActorFrame with the actors as children.
+	local modules = {}
+	local files = FILEMAN:GetDirListing(THEME:GetCurrentThemeDirectory().."Modules/")
+	for file in ivalues(files) do
+		-- Get the file extension (everything past the last period).
+		local filetype = file:match("[^.]+$"):lower()
+		if filetype == "lua" then
+			local full_path = THEME:GetCurrentThemeDirectory().."Modules/"..file
+			Trace("Loading module: "..full_path)
+
+			-- Load the Lua file as proper lua.
+			local loaded_module, error = loadfile(full_path)
+			if loaded_module then
+				local loaded_t = loaded_module()
+				for screenName, actor in pairs(loaded_t) do
+					if modules[screenName] == nil then
+						modules[screenName] = {}
+					end
+					modules[screenName][#modules[screenName]+1] = actor
+				end
+			else
+				Trace("Error loading module: "..full_path.." with error:\n    "..error)

```

Comment:
Maybe use `Warn()` instead of `Trace()`, so the user is more likely to see the error.

Reply:
`lua.ReportScriptError()` is also a viable option here.  It's what I (eventually) used for error-reporting when loading NoteSkin previews.
https://quietly-turning.github.io/Lua-For-SM5/LuaAPI#Namespaces-lua-ReportScriptError

Reply:
Oooh I like lua.ReportScriptError(). I went with that. It does seem like the errors aren't reported though even if ShowThemeErrors is turned on but maybe that might have to do with ScreenSystemLayer is loaded really early on? I'm not sure, but it does make the errors more easily readable in the logs.